### PR TITLE
[HBHE] Remove Phase 1 HE from HBHE noise filter consideration

### PR DESCRIPTION
**Pull request by jaehyeok**
With full depths in Phase 1 HE we found that some variables used in the HBHE noise filter decision need to be updated. 

Phase 1 HE (SiPM+QIE11) has more channels than phase 0 HE, _i.e._, 46 vs 18 channels per RM. Some of the variables in HBHE noise filters use rechit multiplicity per RM, so cut values should be updated accordingly to account for the increase of rechit multiplicity. But, we know that the QIE11 channels don’t have coherent noise [1], so we decided to exclude HE channels in the construction of filter variables; we use only HB channels to calculate the variables to cut on.

We use `HBHERecHitAuxSetter::OFF_TDC_TIME` bit in `HBHERecHit.auxPhase1()` to exclude QIE11 channels because the bit is set to 1 for QIE11 channels and to 0 for QIE8 channels. 

This update will have a visible effect in MC reconstructed with full HE depth segmentation and a small effect in data. 

More details can be found in [2]. 

[1] https://twiki.cern.ch/twiki/pub/CMSPublic/HcalDPGResultsCMSDPS2017042/HEP17noise.pdf
[2] https://indico.cern.ch/event/708228/contributions/2907553/attachments/1605989/2548603/20180223_Jae_HCAL_RemoveHEForNoise.pdf

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by deguio** [comment]
thanks @jaehyeok 
I think it is worth backporting this to 10_0_x 
could you do it?
thanks!

**Comment by slava77** [comment]
@deguio 
why would we need this in 10_0_X?

**Comment by jaehyeok** [comment]
@deguio @mariadalfonso 

Backport to 10_0_X created: https://github.com/cms-sw/cmssw/pull/22363

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/22337	7c7b2a1
- changes are in line with the description and affect only HCAL noise flags where phase-1 (HE) is present
- jenkins tests pass and comparisons with the baseline show differences only in hcalnoise related variables
    - changes are relatively small in the 2017 workflows (HEP17 only)
    - changes are relatively large in the 2018 workflows (in rough agreement with the slides provided with the PR description).

**Comment by slava77** [comment]
hold

pending understanding if this needs to be configurable.
See more in the 100X discussion leading to https://github.com/cms-sw/cmssw/pull/22363#issuecomment-369607653

**Comment by slava77** [comment]
[xposting here as well]
@jaehyeok can we use HBHERecHitAuxSetter::OFF_COMBINED and have the removal only if this bit is not set?

**Comment by jaehyeok** [comment]
@slava77 

Yes, that is a good idea and I will make an update.

A question for @abdoulline about bit "HBHERecHitAuxSetter::OFF_COMBINED". Is this bit always set correctly?

**Comment by abdoulline** [comment]
@jaehyeok  at the first glance (looking at its setting and usage)  no reasons to think it's not. 
But Igor @igv4321 may be aware of some caveats?

**Comment by slava77** [comment]
unhold

File: RecoMET/METProducers/src/HcalNoiseInfoProducer.cc [hunk 1]
```diff
@@ -903,7 +904,11 @@ HcalNoiseInfoProducer::fillrechits(edm::Event& iEvent, const edm::EventSetup& iS
       summary.rechitCount15_ = summary.rechitCount15_ + 1;
       summary.rechitEnergy15_ = summary.rechitEnergy15_ + rechit.eraw();
     }
-
+    
+    // Exclude uncollapsed QIE11 channels
+    if(((rechit.auxPhase1()>>HBHERecHitAuxSetter::OFF_TDC_TIME)&1) &&
+       ((rechit.auxPhase1()>>HBHERecHitAuxSetter::OFF_COMBINED)&0) ) continue;
```
Comment (slava77) [inline RecoMET/METProducers/src/HcalNoiseInfoProducer.cc]: isn't ```anything  & 0 ``` equal to 0 ?

**Comment by igv4321** [comment]
Yes, HBHERecHitAuxSetter::OFF_COMBINED should always be set correctly.

There are some convenience functions in the "DataFormats/HcalRecHit/interface/CaloRecHitAuxSetter.h" header file ("getBit" in particular) for parsing contents of uint32_t status words.

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoMET/METProducers/src/HcalNoiseInfoProducer.cc (hunk 1 above)
Comment (jaehyeok) [reply RecoMET/METProducers/src/HcalNoiseInfoProducer.cc]: You are right. This is fixed in the new commit.

**Comment by jaehyeok** [comment]
@igv4321 

Thanks. I updated the code with CaloRecHitAuxSetter::getBit function.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/22337	0b3598e
- improved since the last signoff https://github.com/cms-sw/cmssw/pull/22337#issuecomment-369409442 to disable the noise filter only for uncollapsed (default in reco) phase-1 HBHE
- jenkins tests pass and comparisons with the baseline show differences only in 2018/2019 tests (the changes previously present in the 2017 are gone as expected for the collapsed reco scenario of the plan-1)

**Comment by fabiocos** [comment]
@slava77 @perrotta in my understanding this PR changes the RECO results at low level for HCAL compared to 10_0_X. Do I understand that anyway this is a needed input to the tuning of high level reconstruction, to be added on top of the 10_0_X RECO? And its backport to 10_0_X is also required for HLT tuning?

**Comment by slava77** [comment]
> @slava77 @perrotta in my understanding this PR changes the RECO results at low level for HCAL compared to 10_0_X. Do I understand that anyway this is a needed input to the tuning of high level reconstruction, to be added on top of the 10_0_X RECO? And its backport to 10_0_X is also required for HLT tuning?

There is no backport to 10_0_X anymore.
The changes in this PR do not affect HCAL hit reconstruction, which could affect the planned calibrations from 10_1_X.

**Comment by fabiocos** [comment]
@slava77 ok, thanks for the clarification

**Comment by fabiocos** [comment]
+1

**Pull request merged**